e2e-tests/cypress/integration/appeal-submission-appeal-site-address/appeal-submission-appeal-site-address.js

```diff
@@ -0,0 +1,122 @@
+  const value = Array(count + 1).join('x');
+
+  switch (component) {
+    case 'Address Line 1':
+      cy.provideAddressLine1(value);
+      break;
+    case 'Address Line 2':
+      cy.provideAddressLine2(value);
+      break;
+    case 'Town or City':
+      cy.provideTownOrCity(value);
+      break;
+    case 'County':
+      cy.provideCounty(value);
+      break;
+    case 'Postcode':
+      cy.providePostcode(value);
+      break;
+  }
+  cy.clickSaveAndContinue();
+});
+
+When(
+  'the user provides values that are too long for Address Line 1, Address Line 2, Town or City, County and Postcode',
+  () => {
+    const value = Array(70).join('x');
+    cy.provideAddressLine1(value);
+    cy.provideAddressLine2(value);
+    cy.provideTownOrCity(value);
+    cy.provideCounty(value);
+    cy.providePostcode(value);
+    cy.clickSaveAndContinue();
+  },
+);
+
+When(
+  'the user provides values that are too long for Address Line 2 and Town or City and provides no other data',
+  () => {
+    const value = Array(70).join('x');
+    cy.provideAddressLine2(value);
+    cy.provideTownOrCity(value);
+    cy.clickSaveAndContinue();
+  },
+);
+
+When('the user provides their appeal site address with postcode as {string}', (postcode) => {
+  cy.providePostcode(postcode);
+  cy.clickSaveAndContinue();
+});
+
+Then('the user is able to continue with the provided address', () => {
+  cy.confirmSiteAddressWasAccepted();
+});
+
+Then(
+  'the user is informed that they cannot continue with the provided address because {string}',
+  (reason) => {
+    switch (reason) {
+      case 'Address Line 1 is required':
+        cy.confirmSiteAddressWasRejectedBecause('Enter a building and/or street');
+        break;
+      case 'County is required':
+        cy.confirmSiteAddressWasRejectedBecause('Enter a county');
+        break;
+      case 'Postcode is required':
+        cy.confirmSiteAddressWasRejectedBecause('Enter a postcode');
+        break;
+      case 'Address Line 1 has a limit of 60 characters':
+        cy.confirmSiteAddressWasRejectedBecause(
+          'Building and/or street must be 60 characters or fewer',
+        );
+        break;
+      case 'Address Line 2 has a limit of 60 characters':
+        cy.confirmSiteAddressWasRejectedBecause(
+          'Building and/or street must be 60 characters or fewer',
+        );
+        break;
+      case 'Town or City has a limit of 60 characters':
+        cy.confirmSiteAddressWasRejectedBecause('Town or city must be 60 characters or fewer');
+        break;
+      case 'County has a limit of 60 characters':
+        cy.confirmSiteAddressWasRejectedBecause('County must be 60 characters or fewer');
+        break;
+      case 'Postcode has a limit of 8 characters':
+        cy.confirmSiteAddressWasRejectedBecause('Postcode must be 8 characters or fewer');
+        break;
+    }
+  },
+);
+
+Then('the user is informed that {string}', (reason) => {
+  switch (reason) {
```

Comment:
is it worth having a `default` case that `throw`s if the given reason isn't one of the expected reasons? 

idk if it is, just a thought

Reply:
Hmm... maybe to force a failure

Reply:
Really nice catch, I found an untested test !